src/state/slices/opportunitiesSlice/selectors.ts

```diff
@@ -141,22 +141,26 @@ export const selectAggregatedUserStakingOpportunityByStakingId = createDeepEqual
 export const selectHighestBalanceAccountIdByStakingId = createSelector(
   selectUserStakingOpportunitiesById,
   selectStakingIdParamFromFilter,
-  (userStakingOpportunities, stakingId): AccountId => {
+  (userStakingOpportunities, stakingId): AccountId | null => {
     if (stakingId === '') return '*' // Narrowing flavoured type
 
     const userStakingOpportunitiesEntries = Object.entries(userStakingOpportunities) as [
       UserStakingId,
       UserStakingOpportunity,
     ][]
-    const [foundUserStakingId] = userStakingOpportunitiesEntries
+    const foundEntry = (userStakingOpportunitiesEntries ?? [])
       .filter(([userStakingId]) =>
         filterUserStakingIdByStakingIdCompareFn(userStakingId, stakingId),
       )
       .sort(([, userStakingOpportunityA], [, userStakingOpportunityB]) =>
         bnOrZero(userStakingOpportunityB.stakedAmountCryptoPrecision)
           .minus(userStakingOpportunityA.stakedAmountCryptoPrecision)
           .toNumber(),
-      )[0]
+      )?.[0]
+
+    const foundUserStakingId = foundEntry?.[0]
+
+    if (!foundUserStakingId) return null
```

Comment:
Just making things safu-er for the runtime

Reply:
As it turns out, we actually *needed* this, this is not only paranoia, see https://github.com/shapeshift/web/commit/2610699428e3f9bfdb80793141bebc13ae96d35a